### PR TITLE
Cumsum cumprod

### PR DESCRIPTION
**Pull request by filipeclduarte**
Following the issue: https://github.com/mratsim/Arraymancer/issues/491 @HugoGranstrom helped me develop those two procedures: `cumsum` and `cumprod`, which are very useful to scientific computing and finance.

### CODE REVIEW

**Comment by mratsim** [comment]
Thanks! Can you add a test in the test suite?

Ultimately we likely want that parallelized and a test would be helpful to ensure that we don't mess it up.

**Comment by filipeclduarte** [comment]
I added the test for these procedures. I'm gonna commit the changes.

File: src/arraymancer/tensor/aggregate.nim [hunk 1]
```diff
@@ -245,3 +245,33 @@ proc iqr*[T](t: Tensor[T]): float =
...
   result = percentile(tS, 75, isSorted = true) -
            percentile(tS, 25, isSorted = true)
+
+proc cumsum*[T](t: Tensor[T], axis:int): Tensor[T] = # from hugogranstrom
+  ## Calculates the cumulative sum of a rank-n Tensor.
+  ## Inputs:
+  ##  - t: a rank-n tensor to cumulatively sum
+  ##  - axis: int
+  ## Returns:
+  ##  - A tensor cumulatively summed at axis, that is, add each value to
```
Comment (mratsim) [inline src/arraymancer/tensor/aggregate.nim:255]: can you add `mixin _` here, it seems like the `_` symbol is not visible in some cases and breaks CI

File: src/arraymancer/tensor/aggregate.nim [hunk 2]
```diff
@@ -245,3 +245,33 @@ proc iqr*[T](t: Tensor[T]): float =
...
+    else:
+      temp[_] = result.atAxisIndex(axis, i-1) + tAxis
+
+proc cumprod*[T](t: Tensor[T], axis:int): Tensor[T] = # from hugogranstrom
+  ## Calculates the cumulative sum of a rank-n Tensor.
+  ## Inputs:
+  ##  - t: a rank-n tensor to cumulatively sum
+  ##  - axis: int
+  ## Returns:
+  ##  - A tensor cumulatively summed at axis, that is, add each value to
```
Comment (mratsim) [inline src/arraymancer/tensor/aggregate.nim:270]: and similar here can you add `mixin _` here, it seems like the `_` symbol is not visible in some cases and breaks CI

**Review by mratsim — APPROVED** [review]
Nim's early symbol resolution strikes again :/ https://github.com/nim-lang/Nim/issues/8677 https://github.com/nim-lang/Nim/issues/6387

File: src/arraymancer/tensor/aggregate.nim (hunk 1 above)
Comment (filipeclduarte) [reply src/arraymancer/tensor/aggregate.nim:255]: added

File: src/arraymancer/tensor/aggregate.nim (hunk 2 above)
Comment (filipeclduarte) [reply src/arraymancer/tensor/aggregate.nim:271]: added

**Pull request merged**